### DISCUSSION TITLE
Understanding fly_to() and set_view_state()

### DISCUSSION THREAD

**Discussion by z-clement** [developmentseed/lonboard#517] [General]
Hi,

I'm working with lonboard for some large spatial data visualizations, and it's been a great tool so far! I'm trying to add a `fly_to()` method to change the view state of the map. I pass in `longitude`, `latitude`, and `zoom` to `fly_to()` and the map view state changes, but not in the way I'd expect.

The basemap changes views to the correct location, however the data don't update, instead it's fixed on the screen, seemingly drawn in the same exact way as when the map first loaded. Additionally, the map stops being interactive after that call, and I get thrown the following error: `TypeError: ViewState.__new__() got an unexpected keyword argument 'transitionDuration'`.

I've tried using `set_view_state()` as well, and while it doesn't throw an error, my plotted data completely disappear after the basemap changes, and it stops being interactive.

Is this a bug, or am I not understanding the use cases of these two functions? Thanks for any insight! (Also am unsure if discussions is the place for this, can repost elsewhere if that fits better.)

**Comment by kylebarron** [comment]
This sounds like a bug so moving to an issue: https://github.com/developmentseed/lonboard/issues/518
